sympy/functions/elementary/exponential.py

```diff
@@ -495,8 +495,13 @@ def _eval_nseries(self, x, n, logx, cdir=0):
             return Order(x**n, x)
         if arg0 is S.Infinity:
             return self
+        for arg1 in arg.args:
+            if isinstance(arg1, log):
+                term = arg/arg1
+                if not (term).has(x) and exp(arg1).has(x) and term.is_number:
+                    return self
         # checking for indecisiveness/ sign terms in arg0
-        if any(isinstance(arg, (sign, ImaginaryUnit)) for arg in arg0.args):
+        if any(isinstance(arg, sign) for arg in arg0.args):
```

Comment:
I would try something like this 
```
--- a/sympy/functions/elementary/exponential.py
+++ b/sympy/functions/elementary/exponential.py
@@ -495,8 +495,10 @@ def _eval_nseries(self, x, n, logx, cdir=0):
             return Order(x**n, x)
         if arg0 is S.Infinity:
             return self
+        if arg0.is_infinite and arg.subs(x, 0).has(S.NaN, S.ComplexInfinity):
+            raise PoleError("Cannot expand %s around 0" % (self))
         # checking for indecisiveness/ sign terms in arg0
-        if any(isinstance(arg, (sign, ImaginaryUnit)) for arg in arg0.args):
+        if any(isinstance(arg, (sign)) for arg in arg0.args):
             return self
         t = Dummy("t")
```
Maybe we won't even require the `arg.subs(x, 0).has(S.NaN, S.ComplexInfinity)` check. Please try it out and check if we need that.

Reply:
Yeah, I beileve you are right that we need not check for `arg.subs(x, 0).has(S.NaN, S.ComplexInfinity)` , so we can move on